contrib/completion/_fd

```diff
@@ -0,0 +1,210 @@
+    '!--xdev'
+
+    + dir-depth # directory depth
+    '(--exact-depth -d --max-depth)'{-d+,--max-depth=}'[set max directory depth to descend when searching]:depth'
+    '!(--exact-depth -d --max-depth)--maxdepth:depth'
+    '(--exact-depth --min-depth)--min-depth=[set directory depth to descend before start searching]:depth'
+    '(--exact-depth -d --max-depth --maxdepth --min-depth)--exact-depth=[only search at the exact given directory depth]:depth'
+
+    + filter-misc # Filter search
+    '*'{-t+,--type=}"[filter search by type]:type:(($fd_file_types))"
+    '*'{-e+,--extension=}'[filter search by file extension]:extension'
+    '*'{-E+,--exclude=}'[exclude files/directories that match the given glob pattern]:pattern'
+    '*'{-S+,--size=}'[limit search by file size]:size limit'
+    '(-o --owner)'{-o+,--owner=}'[filter by owning user and/or group]:owner/group'
+
+    + ignore-file
+    '*--ignore-file=[add a custom, low-precedence ignore-file with .gitignore format]: :_files'
+
+    + '(filter-mtime-newer)' # Filter files modified after than
+    '--changed-within=[limit search to files/directories modified within the given date/duration]:date/duration'
+    '!--change-newer-than=:date/duration'
+    '!--newer=:date/duration'
+
+    + '(filter-mtime-older)' # Filter files modified before than
+    '--changed-before=[limit search to files/directories modified before the given date/duration]:date/duration'
+    '!--change-older-than=:date/duration'
+    '!--older=:date/duration'
+
+    + '(color)' # all other options
+    {-c+,--color=}'[declare when to colorize search results]:when to colorize:((
+      auto\:"show colors if the output goes to an interactive console"
+      never\:"do not use colorized output"
+      always\:"always use colorized output"
+    ))'
+
+    + '(threads)'
+    {-j+,--threads=}'[set the number of threads for searching and executing]:threads'
+
+    + '(exec-cmds)' # execute command
+    '(long-listing max-results)'{-x+,--exec=}'[execute command for each search result]:command: _command_names -e:*\;::program arguments: _normal'
+    '(long-listing max-results)'{-X+,--exec-batch=}'[execute command for all search results at once]:command: _command_names -e:*\;::program arguments: _normal'
+
+    + other
+    $no'(--max-buffer-time)--max-buffer-time=:milliseconds'
+
+    + '(about)' # about flags
+    '(: * -)'{-h,--help}'[display help message]'
+    '(: * -)'{-v,--version}'[display version information]'
+
+    + search-path
+    $no'(--base-directory)--base-directory=[change the current working directory to the given path]:directory:_files -/'
+    $no'(*)*--search-path=:directory:_files -/'
+
+    + args # positional arguments
+    '1: :_guard "^-*" pattern'
+    '*:directory:_files -/'
+  )
+
+  # Strip out argument groups where unsupported (see above)
+  is-at-least 5.4 ||
+  fd_args=( ${(@)args:#(#i)(+|[a-z0-9][a-z0-9_-]#|\([a-z0-9][a-z0-9_-]#\))} )
+
+  _arguments $_arguments_options : $fd_args && ret=0
```

Comment:
The `ret=` and `return` stuff here is redundant; since `_arguments` is the last command in the function its return value will just fall through. The reason many zsh completion functions do this `ret=` thing is that they need it to ensure that state-handling code passes back the correct status without immediately returning; you don't have any state handling here, so it's superfluous. Though if you expect to add that some day, maybe that's an argument for keeping it, i suppose

Reply:
I originally had it that way, without `ret`. But as you said, I though that in the future someone may add more logic to the function, like a `case $state in`, and may not know about the `ret` idiom, so I added it to make sure it is noticed.